### PR TITLE
Fix nova session

### LINKED ISSUE #138: nova session is not working after rewrite
https://github.com/novaframework/nova/blob/fix-nova-session/src/nova_cache.erl#L107

On this line we try to get a cache, but the Cache is not an atom so we get an function clause.

My branch above solved that we was calling a module called nova_session_cache that have been renamed nova_cache.

### PR DESCRIPTION
**Pull request by Taure**
close #138

### CODE REVIEW

File: src/nova_cache.erl [hunk 1]
```diff
@@ -104,8 +104,8 @@ get(Cache, Key) ->
 %%--------------------------------------------------------------------
 -spec set(Cache :: atom(), Key :: term(), Value :: any(), TTL :: non_neg_integer()) -> ok.
 set(Cache, Key, Value, TTL) ->
-    {ok, Id} = nova_cache_sup:get_cache(Cache),
-    gen_server:call(Id, {set, Key, Value, TTL}).
+    {ok, Id} = nova_cache_sup:get_cache(Key),
+    gen_server:call(Id, {set, Cache, Value, TTL}).
```
Comment (burbas) [inline src/nova_cache.erl:108]: I think this got a bit mixed up. We should query the cache (Based on the cache-value) with Key and Value.

File: src/nova_session.erl [hunk 2]
```diff
@@ -86,7 +86,7 @@ delete(Req, Key) ->
     case get_session_id(Req) of
         {ok, SessionId} ->
             Mod = get_session_module(),
-            Mod:delete_value(SessionId, Key),
+            Mod:delete(SessionId, Key),
```
Comment (burbas) [inline src/nova_session.erl:89]: This should be delete_value. It uses the callback-functions for operations like this.

Comment (burbas) [reply src/nova_session.erl:89]: This goes for all changes in this file.

File: src/nova_session.erl [hunk 3]
```diff
@@ -103,7 +103,7 @@ get_session_module() ->
             Module;
         _ ->
             %% Default to nova_session_ets
-            nova_session_cache
+            nova_cache
```
Comment (burbas) [inline src/nova_session.erl:106]: This should be `nova_session_ets`. The nova_cache is used for other use-cases :)

**Review by burbas — CHANGES_REQUESTED** [review]

**Comment by burbas** [comment]
I think it's been a confusion around session vs cache. They are two separate things and used for different purposes. nova_session is only for handling session-data while cache can be used to store intermediate data.

I think we might scrap the cache in favor of the session-modules, but need to take a closer look at what they are used for.

File: src/nova_cache.erl (hunk 1 above)
Comment (Taure) [reply src/nova_cache.erl:108]: But what is the cache?

Is it a key or how is it created?

**Pull request closed without merging**